compendium/ServiceComponent/include/cppmicroservices/servicecomponent/detail/ComponentInstanceImpl.hpp

```diff
@@ -455,17 +455,17 @@ namespace cppmicroservices
                               class I = Injection,
                               class InjectionTrue = typename std::enable_if<I::value == true>::type,
                               class HasNoConstructorWithReferences = typename std::enable_if<
-                                  std::is_constructible<C, std::shared_ptr<CtorInjectedRefs> const&...>::value
+                                  std::is_constructible<C, CtorInjectedRefs const&...>::value
                                   == false>::type,
                               class HasNoConstructorWithRefAndConfig = typename std::enable_if<
                                   std::is_constructible<C,
                                                         std::shared_ptr<cppmicroservices::AnyMap> const&,
-                                                        std::shared_ptr<CtorInjectedRefs> const&...>::value
+                                                        CtorInjectedRefs const&...>::value
                                   == false>::type>
                     std::shared_ptr<T>
                     DoCreate(bool const&)
                     {
-                        static_assert(std::is_constructible<C, std::shared_ptr<CtorInjectedRefs> const&...>::value,
```

Comment:
I think removing the `std::shared_ptr<>` here is going to be a problem when detecting failure modes in service constructors at build time. There is a requirement (prior to the introduction of multiple cardinality) that all injected service references be passed into service constructors in the order they appear in the manifest.json and by `std::shared_ptr<T> const&`. This requirement still holds with multiple cardinality and I think this change no longer detects a violation of this requirement. You should be able to create a negative test case which should exercise this `static_assert`.

Reply:
I modified existing test case for reference order to accommodate these changes - TestCompInst_RefOrder file.